# Faris Abu Farha
# 1200546

# 0. Import the necessary libraries

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt


## 1-Read the dataset and examine how many features and examples does it have?

In [3]:
df = pd.read_csv('data/cars.csv')
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 398 entries, 0 to 397
Data columns (total 8 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   mpg           398 non-null    float64
 1   cylinders     398 non-null    int64  
 2   displacement  398 non-null    float64
 3   horsepower    392 non-null    float64
 4   weight        398 non-null    int64  
 5   acceleration  398 non-null    float64
 6   model_year    398 non-null    int64  
 7   origin        396 non-null    object 
dtypes: float64(4), int64(3), object(1)
memory usage: 25.0+ KB


There are 8 features and 398 examples

## 2- Are there features with missing values? How many missing values are there in each one?

In [12]:
print(df.isnull().sum())

mpg             0
cylinders       0
displacement    0
horsepower      6
weight          0
acceleration    0
model_year      0
origin          2
dtype: int64


from this output we can see how many missing values are there in each feature

## 3- Fill the missing values in each feature using a proper imputation method (for
example: fill with mean, median, or mode) 

In [13]:
df_mean_filled = df.copy()
df_median_filled = df.copy()